### PR TITLE
feat: Add Issue section to PR template

### LINKED ISSUE #4113: [Bug]: Improve pull request template
### Actual Behavior

When I was reviewing the checklist in the form for my PR #4112, I went to the Description section to link the issue that corresponds to my contribution. Even though it the checkbox briefly says where to link the issue, this can still cause new open source contributors to have a hard time create an effective PR because they tend to be not familiar with how to format their links. 

### Expected Behavior

I anticipated that the Issue section would appear in the Description as mentioned.

## Proposed Solution
I would like to add an issue section underneath the Description section of the `PULL_REQUEST_TEMPLATE.md` file. Doing this would make it easier for contributors to link the issues that are addressed in their Pull Requests. 

### On what browser(s) are you seeing the problem?

Chrome

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by CBID2**
### Description
This PR adds an `Issue` section with commented instructions on how contributors should link the issue numbers that their contribution addresses and a revised version of the last checkbox’s sentence. By adding this section, contributors know how to create effective PRs.  

### Issue
Closes #4113 

### Type of Change

- Updating the documentation

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -15,10 +15,17 @@ If you would like to receive credit for your contribution to an entry, make sure
 Of course, you can opt not to do this and be listed as an "Anonymous contributor", instead. :)
 -->
 
-### Description
+# Description
```
Comment (CBID2) [inline .github/PULL_REQUEST_TEMPLATE.md:18]:
```suggestion
### Description
```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -15,10 +15,17 @@ If you would like to receive credit for your contribution to an entry, make sure
...
 
 <!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->
 
+## Issue Solved
```
Comment (CBID2) [inline .github/PULL_REQUEST_TEMPLATE.md:22]:
```suggestion
### Issue Solved
```

**Comment by CBID2** [comment]
Hi @KTom101! :) I'm ready for your review! :)

**Comment by ishg-153** [comment]
Hey @CBID2! Will this edit include instructions in the comments as mentioned in the description?

**Comment by CBID2** [comment]
> Hey @CBID2! Will this edit include instructions in the comments as mentioned in the description?

Yes @ishg-153

**Comment by ishg-153** [comment]
> Yes @ishg-153

Are you yet to add that, because I can't see that change? I see your edit to the last checklist item.

**Comment by CBID2** [comment]
> > Yes @ishg-153
> 
> 
> 
> @CBID2 Are you yet to add that, because I can't see that change? I see your edit to the last checklist item.

Hi @ishg-153. I added them.

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 3]
```diff
@@ -19,6 +19,13 @@ Of course, you can opt not to do this and be listed as an "Anonymous contributor
 
 <!-- Please write a summary of the change, such as which topic(s) and file(s) that you have edited or created. Please also include relevant motivation and context: -->
 
+### Issue Solved
+
+<!--
+Please use the following format(s) to link the issue numbers that your contribution addresses: Fixes #issue number(e.g., #123)
+Relates to #issue number(e.g., #123)
```
Comment (ishg-153) [inline .github/PULL_REQUEST_TEMPLATE.md:26]:
```suggestion
Please use the following format(s) to link the issue numbers that your contribution addresses: 
Closes #issue-number (where `issue-number` corresponds to the PRs' respective issue)
```

**Review by ishg-153 — CHANGES_REQUESTED** [review]
@CBID2 I've suggested one change, please make it.

**Comment by CBID2** [comment]
> @CBID2 I've suggested one change, please make it.

Done @ishg-153

**Review by ishg-153 — APPROVED** [review]
@CBID2 This looks good to me. Ready for a second review!

**Review by avdhoottt — APPROVED** [review]
LGTM 🚀

**Pull request merged**